### PR TITLE
Append gdk version to client user agent string

### PR DESCRIPTION
**Pull request by jkauffman1**
Reformat user agent string
    
    1) Enforce a maximum length of 64 characters (server will reject
       anything more)
    2) Use the short git commit id of the gdk version for brevity
    3) Append the client provided string to the gdk git commit id rather
       than replacing it

### CODE REVIEW

File: src/ga_session.cpp [hunk 1]
```diff
@@ -347,7 +347,7 @@ namespace sdk {
         , m_is_locked(false)
         , m_tx_last_notification(std::chrono::system_clock::now())
         , m_cache(m_net_params, net_params.at("name"))
-        , m_user_agent(net_params.value("user_agent", GDK_COMMIT))
+        , m_user_agent(net_params.value("user_agent", "") + " " + GDK_COMMIT)
         , m_electrum_url(
```
Comment (JamieDriver) [inline src/ga_session.cpp:351]: If there is no 'user_agent' this will result in a leading space before the gdk commit/version - is that ok ?

**Comment by jgriffiths** [comment]
utack a1e5ec3f3f0ec66c6f68d57a5783cf7ad546d535 if we add a runtime assert int the ctor body that the user agent is under some sensible length.

File: src/ga_session.cpp (hunk 1 above)
Comment (jkauffman1) [reply src/ga_session.cpp:351]: It will yes I think it's probably ok for simplicity?

**Comment by jkauffman1** [comment]
> utack [a1e5ec3](https://github.com/Blockstream/gdk/commit/a1e5ec3f3f0ec66c6f68d57a5783cf7ad546d535) if we add a runtime assert int the ctor body that the user agent is under some sensible length.

Edited to warn and truncate to 64 characters

File: src/ga_session.cpp [hunk 2]
```diff
@@ -347,7 +354,7 @@ namespace sdk {
         , m_is_locked(false)
         , m_tx_last_notification(std::chrono::system_clock::now())
         , m_cache(m_net_params, net_params.at("name"))
-        , m_user_agent(net_params.value("user_agent", GDK_COMMIT))
+        , m_user_agent(GDK_COMMIT + " " + net_params.value("user_agent", ""))
```
Comment (jgriffiths) [inline src/ga_session.cpp:357]: need to substring the user provided user agent to 20 chars IMO

**Pull request merged**